rust/arrow/src/array/transform/primitive.rs

```diff
@@ -0,0 +1,37 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use std::{io::Write, mem::size_of};
+
+use crate::{array::ArrayData, datatypes::ArrowNativeType};
+
+use super::{Extend, _MutableArrayData};
+
+pub(super) fn build_extend<T: ArrowNativeType>(array: &ArrayData) -> Extend {
+    let values = &array.buffers()[0].data()[array.offset() * size_of::<T>()..];
+    Box::new(
+        move |mutable: &mut _MutableArrayData, start: usize, len: usize| {
+            let start = start * size_of::<T>();
```

Comment:
it seems to me that `size_of::<T>` is the part here that is dependent on type -- maybe you could just use that as a number

Reply:
> I try to use generics for things that are known at compile time and arguments for things that are only known at runtime.

I agree that is a good general rule of thumb. 

I guess this was a second way of asking if we could avoid dynamic dispatch - clearly the answer was "no" :)